bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/GatewayCommand.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.openthermgateway.internal;
+
+import java.util.HashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class GatewayCommand {
+    private String code;
+    private String validationSet;
+    private String message;
+
+    public String getCode() {
+        return code;
+    }
+
+    public String getMessage() {
+        return this.message;
+    }
+
+    public String getValidationSet() {
+        return validationSet;
+    }
+
+    public String toFullString() {
+        return this.code + "=" + this.message;
+    }
+
+    private GatewayCommand(String code, String message, String validationSet) throws Exception {
+        this.code = code;
+        this.message = message;
+        this.validationSet = validationSet;
+
+        if (!validate()) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid value '%s' for code '%s'", this.message, this.code));
+        }
+    }
+
+    private boolean validate() {
+        if (this.validationSet == null || this.validationSet == "") {
+            return true;
+        }
+
+        String[] validations = this.validationSet.split(",");
+
+        for (int i = 0; i < validations.length; i++) {
+            if (this.message.equals(validations[i])) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static GatewayCommand parse(String code, String message) throws Exception {
```

Comment:
Can you make the thrown exception more specific or does a method actual throw Exception?

Reply:
Changed to IllegalArgumentException, and with that, also the try/catch in calling code